torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2110,7 +2191,8 @@ def set_state_dict_type(
         module: nn.Module,
         state_dict_type: StateDictType,
         state_dict_config: Optional[StateDictConfig] = None,
-    ) -> Tuple[StateDictType, StateDictConfig]:
+        optim_state_dict_config: Optional[OptimStateDictConfig] = None,
+    ) -> Tuple[StateDictType, StateDictConfig, OptimStateDictConfig]:
```

Comment:
unrelated to this change - does `set_state_dict_type` need to be public or private?